src/common/lttng-elf.c

```diff
@@ -0,0 +1,1035 @@
+		ERR("Cannot get ELF stap note section data.");
+		ret = LTTNG_ERR_ELF_PARSING;
+		goto destroy_elf_error;
+	}
+
+	curr_data_ptr = stap_note_section_data;
+	next_note_ptr = stap_note_section_data;
+	curr_note_section_begin = stap_note_section_data;
+
+	*offsets = NULL;
+	while (1) {
+		curr_data_ptr = next_note_ptr;
+		/* Check if we have reached the end of the note section. */
+		if (curr_data_ptr >=
+				curr_note_section_begin + stap_note_section_hdr->sh_size) {
+
+			*nb_probe = nb_match;
+			*offsets = probe_locs;
+			ret = 0;
+			break;
+		}
+		/* Get name size field. */
+		name_size = next_4bytes_boundary(*(uint32_t*) curr_data_ptr);
+		curr_data_ptr += sizeof(uint32_t);
+
+		/* Sanity check; a zero name_size is reserved. */
+		if (name_size == 0) {
+			ERR("Invalid name size field in SDT probe descriptions"
+				"section.");
+			ret = -1;
+			goto realloc_error;
+		}
+
+		/* Get description size field. */
+		desc_size = next_4bytes_boundary(*(uint32_t*) curr_data_ptr);
+		curr_data_ptr += sizeof(uint32_t);
+
+
+		/* Get type field. */
+		note_type = *(uint32_t *) curr_data_ptr;
+		curr_data_ptr += sizeof(uint32_t);
+
+		/*
+		 * Move the pointer to the next note to be ready for the next
+		 * iteration. The current note is made of 3 unsigned 32bit
+		 * integers (name size, descriptor size and note type), the
+		 * name and the descriptor. To move to the next note, we move
+		 * the pointer according to those values.
+		 */
+		next_note_ptr = next_note_ptr +
+			(3 * sizeof(uint32_t)) + desc_size + name_size;
+
+		/*
+		 * Move ptr to the end of the name string (we don't need it)
+		 * and go to the next 4 byte alignement.
+		 */
+		if (note_type != NOTE_STAPSDT_TYPE ||
+			strncmp(curr_data_ptr, NOTE_STAPSDT_NAME, name_size) != 0) {
+			continue;
+		}
+
+		curr_data_ptr += name_size;
+
+		/* Get description field. */
+		curr_desc_beg = curr_data_ptr;
+
+		/* Get probe location.  */
+		curr_probe_location = *(uint64_t *) curr_data_ptr;
+		curr_data_ptr += sizeof(uint64_t);
+
+		/* Pass over the base. Not needed. */
+		curr_data_ptr += sizeof(uint64_t);
+
+		/* Get semaphore location. */
+		curr_semaphore_location = *(uint64_t *) curr_data_ptr;
+		curr_data_ptr += sizeof(uint64_t);
+		/* Get provider name. */
+		curr_provider = curr_data_ptr;
+		curr_data_ptr += strlen(curr_provider) + 1;
+
+		/* Get probe name. */
+		curr_probe = curr_data_ptr;
+		curr_data_ptr += strlen(curr_probe) + 1;
+
+		curr_data_ptr = curr_desc_beg + desc_size;
+
+		/* Check if the provider and probe name match */
+		if (strcmp(provider_name, curr_provider) == 0 &&
+			strcmp(probe_name, curr_probe) == 0) {
+
+			/*
+			 * We currently don't support SDT probes with semaphores. Return
+			 * success as we found a matching probe but it's guarded by a
+			 * semaphore.
+			 */
+			if (curr_semaphore_location != 0) {
+				ret = LTTNG_ERR_SDT_PROBE_SEMAPHORE;
+				goto end;
+			}
+
+			int new_size = ++nb_match * sizeof(uint64_t);
```

Comment:
Move declaration to the beginning of the condition's scope.

Reply:
done